File: src/main/java/ordilov/lissn/playlist/domain/youtube/YoutubeVideo.java
package ordilov.lissn.playlist.domain.youtube;

import java.util.List;
import lombok.Getter;

@Getter
public class YoutubeVideo {

  private String kind;
  private String etag;
  private List<Item> items;
  private PageInfo pageInfo;

  @Getter
  public static class ContentDetails{
    private String note;
    private String endAt;
    private String startAt;
    private String videoId;
  }

  @Getter
  public static class Statistics{
    private Long viewCount;
    private Long likeCount;
    private Long dislikeCount;
    private Long favoriteCount;
    private Long commentCount;
  }

  @Getter
  public static class Item{
    private String id;
    private String kind;
    private String etag;
    private Snippet.Video snippet;
    private ContentDetails contentDetails;
    private Statistics statistics;
  }
}

File: src/main/java/ordilov/lissn/playlist/domain/youtube/YoutubeVideo.kt
package ordilov.lissn.playlist.domain.youtube

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class YoutubeVideo(

    val kind: String,
    val etag: String,
    val items: List<Item>,
    val pageInfo: PageInfo,
) {
    data class ContentDetails(
        val note: String,
        val endAt: String,
        val startAt: String,
        val videoId: String,
    )

    data class Statistics(
        val viewCount: Long,
        val likeCount: Long,
        val dislikeCount: Long,
        val favoriteCount: Long,
        val commentCount: Long,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Item(
        val id: String,
        val kind: String,
        val etag: String,
        val snippet: Video,
        val contentDetails: ContentDetails,
        val statistics: Statistics,
    )
}